ts-rs/src/export.rs

```diff
@@ -26,6 +29,51 @@ pub enum ExportError {
     ManifestDirNotSet,
 }
 
+/// Exports `T` to the file specified by the `#[ts(export_to = ..)]` attribute.
+/// Additionally, all dependencies of `T` will be exported as well.
+/// TODO: This might cause a race condition:
+///       If two types `A` and `B` are `#[ts(export)]` and depend on type `C`,
+///       then both tests for exporting `A` and `B` will try to write `C` to `C.ts`.
+///       Since rust, by default, executes tests in paralell, this might cause `C.ts` to be corrupted.
+pub(crate) fn export_type_with_dependencies<T: TS + ?Sized + 'static>() -> Result<(), ExportError> {
```

Comment:
I have now added this function, which exports `T` and recursively all of its dependencies.  

The next step here should probably be to actually cause a race condition, and then figure out how to fix it (maybe by putting a `static MUTEX: Mutex<()>` in `TS`? Or run the tests with `RUST_TEST_THREADS=1` to disable paralell execution?)

Reply:
Oh yeah - that fixed both E2E tests ^^

Reply:
Oh yeah that's gonna be a hard one to handle! Especially since the compiler, which usually protects us from race conditions, can't do much for us here in test land

Reply:
Interestingly enough, I cant get it to fail, even with 16 types.
I think one easy solution might be to first write to a temp file, and then rename it. Gotta look into that, but I think renaming is atomic on all platforms.

Reply:
> Oh yeah - that fixed both E2E tests ^^

What fixed it? The mutex or the `RUST_TEST_THREADS`?

Reply:
Nothing! I just tried to provoke the error, by exporting lots of types and setting `RUST_TEST_THREADS=16`, but nothing broke. Maybe that's because the files are relatively small, and get written in just one syscall. I'll try exporting larger types, maybe that'll break it.

Reply:
Yeah, idk. I made the types now 17MB large with 2 million `number`s in them. `cargo test` still refuses to corrupt them..